src/components/CartContent/ThemeContext.tsx

```diff
@@ -0,0 +1,31 @@
+import type { ReactNode } from 'react';
+import React, { createContext, useContext, useState } from 'react';
+
+interface ThemeContextProps {
+    isDarkMode: boolean;
+    onThemeChange: (isDarkMode: boolean) => void;
+}
+
+interface ThemeProviderProps {
+    children: ReactNode;
+}
+
+const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);
```

Comment:
goo practies iniotialise contexts with null or default value, try to use as little as possible of `undefined`'s